pkg/initializer/initializer.go

```diff
@@ -88,13 +169,15 @@ func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int6
 
 	metrics.ValidationDurationSeconds.With(prometheus.Labels{metrics.LabelSucceeded: metrics.ValueSucceededTrue}).Observe(time.Since(start).Seconds())
 
+	// bootstrap of the first member after quorum loss case as well as the bootstrap case come to this point
+	// so, check for quorum loss case and skip restoration using removing and adding member
 	if dataDirStatus != validator.DataDirectoryValid {
-		if dataDirStatus == validator.DataDirStatusInvalidInMultiNode || (e.Validator.OriginalClusterSize > 1 && dataDirStatus == validator.DataDirectoryCorrupt) || (e.Validator.OriginalClusterSize > 1 && isEtcdMemberPresent) {
+		if (dataDirStatus == validator.DataDirStatusInvalidInMultiNode || (e.Validator.OriginalClusterSize > 1 && dataDirStatus == validator.DataDirectoryCorrupt) || (e.Validator.OriginalClusterSize > 1 && isEtcdMemberPresent)) && !quorumLossCaseFound {
```

Comment:
> .... &&`! quorumLossCaseFound ` 

Why we required this condition in case of quorumLoss ?
AFAIU, if there is a quorum loss then controller will scale-down the cluster to 0 then scale-up to 1.
So, IMO  `OriginalClusterSize` will handle this case ... isn't ?

Reply:
Nope. It's not handling

Reply:
Done